### PR TITLE
KYLIN-4336 Global domain dict for MR build engine

### PR DESCRIPTION
**Pull request by wangxiaojing123**
issue link: https://issues.apache.org/jira/browse/KYLIN-4336

### CODE REVIEW

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDesc.java [hunk 1]
```diff
@@ -1375,12 +1376,37 @@ public void setParentForward(int parentForward) {
                     result.remove(dictDesc.getColumnRef());
                     result.add(dictDesc.getResuseColumnRef());
                 }
+
+                //tiretree global domain dic
+                if (Objects.isNull(dictDesc.getResuseColumnRef()) && Objects.nonNull(dictDesc.getReuseColumn())) {
+                    logger.info("tiretree global domain dic : column {} use tiretree global domain dic, reuse column {} ", dictDesc.getColumnRef(), dictDesc.getReuseColumn());
```
Comment (nichunen) [inline core-cube/src/main/java/org/apache/kylin/cube/model/CubeDesc.java]: I think it's better to put this into the scope of the previous conditional expression "if (dictDesc.getResuseColumnRef() != null) "

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+     * add resuce global tiretree global dic for baseid job
+     * @param cubeDesc
+     * @param dumpList
+     */
+    public static void cuboidJob(CubeDesc cubeDesc, Set<String> dumpList) {
+        logger.info("cube {} start to add global domain dic", cubeDesc.getName());
+        CubeManager cubeManager = CubeManager.getInstance(KylinConfig.getInstanceFromEnv());
+        DataModelManager metadataManager =DataModelManager.getInstance(KylinConfig.getInstanceFromEnv());
+
+        cubeManager.getCube(cubeDesc.getName());
```
Comment (nichunen) [inline core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java]: useless

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+
+
+
+    /**
+     * add resuce global tiretree global dic for baseid job
+     * @param cubeDesc
+     * @param dumpList
+     */
+    public static void cuboidJob(CubeDesc cubeDesc, Set<String> dumpList) {
```
Comment (nichunen) [inline core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java]: The method name is not clear enough to clarify its function

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDesc.java (hunk 1 above)
Comment (wangxiaojing123) [reply core-cube/src/main/java/org/apache/kylin/cube/model/CubeDesc.java]: Thanks for coding review .The previous conditional expression use for reuse col of the same cube (Existing functions of kylin), and dictDesc.getResuseColumnRef() is not null, but the new global domain dic's dictDesc.getResuseColumnRef() is null . After the expression is satisfied, the content to be processed is different too， the global domain dic only result.remove(dictDesc.getColumnRef()) no result.add(dictDesc.getResuseColumnRef()).

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java (hunk 2 above)
Comment (wangxiaojing123) [reply core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java]: Thanks ,I will fix it.

File: core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java (hunk 3 above)
Comment (wangxiaojing123) [reply core-cube/src/main/java/org/apache/kylin/cube/model/CubeDescTiretreeGlobalDomainDictUtil.java]: OK, I will fix it, thanks.

**Comment by nichunen** [comment]
Merged in PR https://github.com/apache/kylin/pull/1058

**Pull request closed without merging**